tests/test_edge_cases.py

```diff
@@ -1296,6 +1296,13 @@ class Model(BaseModel):
     }
 
 
+def test_no_name_conflict_in_constructor():
+    class Model(BaseModel):
+        pass
+
+    assert Model(**{'__pydantic_self__': 4})
```

Comment:
Can you also add `self` in your kwargs?
Originally, the main goal of `__pydantic_self__` was to be able to have a field name `self`
When we don't support 3.7, the positional-only solution is way better indeed.

Reply:
Makes sense 👍 , I've added it. Interestingly there are other `self` tests in `test_edge_cases.py`, but they seem to only apply to `model_validate()`, which shouldn't have this problem in the first place 🤔